File: 0539-minimum-time-difference/0539-minimum-time-difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        int n = timePoints.size();
        vector<int> minutes;

        // Converting  HH:MM to minutes
        for (auto &t : timePoints) {
            int h = stoi(t.substr(0, 2));
            int m = stoi(t.substr(3, 2));
            minutes.push_back(h * 60 + m);
        }
        sort(minutes.begin(), minutes.end());

        int minDiff = 1440;
        for (int i = 1; i < n; i++) {
            minDiff = min(minDiff, minutes[i] - minutes[i - 1]);
        }

        // Circular difference (last & first)
        minDiff = min(minDiff, 1440 - minutes[n - 1] + minutes[0]);

        return minDiff;
    }
};

File: 0539-minimum-time-difference/0539-minimum-time-difference.java
import java.util.*;

class Solution {
    public int findMinDifference(List<String> timePoints) {
    
        List<Integer> list = new ArrayList<>();
        
        for (String t : timePoints) {
            String[] parts = t.split(":");
            int hours = Integer.parseInt(parts[0]);
            int minutes = Integer.parseInt(parts[1]);
            list.add(hours * 60 + minutes);
        }
        Collections.sort(list);

    
        int minDiff = 1440;
        
        for (int i = 1; i < list.size(); i++) {
            int diff = list.get(i) - list.get(i - 1);
            minDiff = Math.min(minDiff, diff);
        }

        // Step 4: Check circular difference (last and first)
        int circularDiff = 1440 - list.get(list.size() - 1) + list.get(0);
        minDiff = Math.min(minDiff, circularDiff);

        return minDiff;
    }
}